registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/BioSdkUtil.java

```diff
@@ -0,0 +1,66 @@
+package io.mosip.registration.processor.packet.storage.utils;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import io.mosip.kernel.biometrics.commons.CbeffValidator;
+import io.mosip.kernel.biometrics.constant.BiometricFunction;
+import io.mosip.kernel.biometrics.constant.BiometricType;
+import io.mosip.kernel.biometrics.entities.BIR;
+import io.mosip.kernel.biosdk.provider.factory.BioAPIFactory;
+import io.mosip.kernel.biosdk.provider.spi.iBioProviderApi;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.kernel.core.util.CryptoUtil;
+import io.mosip.registration.processor.core.constant.MappingJsonConstants;
+import io.mosip.registration.processor.core.idrepo.dto.Documents;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+
+@Component
+public class BioSdkUtil {
+
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(BioSdkUtil.class);
+
+	/** The utilities. */
+	@Autowired
+	Utilities utilities;
+
+	/** The bio api factory. */
+	@Autowired(required = false)
+	private BioAPIFactory bioApiFactory;
+
+	/** modality arrays that needs to be tagged */
+	@Value("#{'${mosip.regproc.quality.classifier.tagging.quality.modalities}'.split(',')}")
+	private List<String> modalities;
```

Comment:
this is not correct. See all the modalities can be there inside packet for authenticating biometric. But country may choose to use only few modalities for tagging. This property says which modalities should be tagged. If country doesnot choose all biometrics then your code will not work. You dont need this property, You can read all modalities from the bir object itself in line no 52